rs/boundary_node/anonymization/client/src/lib.rs

```diff
@@ -1 +1,253 @@
+use std::time::SystemTime;
 
+use anonymization_interface::{self as ifc};
+use anyhow::{anyhow, Context};
+use async_trait::async_trait;
+use candid::{Decode, Encode, Principal};
+use ic_canister_client::Agent;
+
+#[derive(Debug, thiserror::Error)]
+pub enum RegisterError {
+    #[error("unauthorized")]
+    Unauthorized,
+
+    #[error(transparent)]
+    UnexpectedError(#[from] anyhow::Error),
+}
+
+#[async_trait]
+pub trait Register: Sync + Send {
```

Comment:
Curios, why do You need `Sync + Send` here. Is there going to be concurrency involved?

Reply:
This is required because `Register` is an async trait.